src/Analyzers/CSharp/Tests/RemoveUnnecessaryCast/RemoveUnnecessaryCastTests.cs

```diff
@@ -3550,20 +4035,38 @@ void Goo(Task<Action> x)
         (x.Result)();
     }
 }
-");
+";
+
+            await new VerifyCS.Test
+            {
+                TestCode = source,
+                FixedState =
+                {
+                    Sources = { fixedSource },
+                    ExpectedDiagnostics =
+                    {
+                        // /0/Test0.cs(9,10): error CS0118: 'x' is a variable but is used like a type
+                        DiagnosticResult.CompilerError("CS0118").WithSpan(9, 10, 9, 11).WithArguments("x", "variable", "type"),
+                        // /0/Test0.cs(9,20): error CS1525: Invalid expression term ')'
+                        DiagnosticResult.CompilerError("CS1525").WithSpan(9, 20, 9, 21).WithArguments(")"),
```

Comment:
⚠️  This case is special: the original source did not have compilation errors, but the code fix transformed it to code that does not compile.